src/spatial/spatial_massaction_jump.jl

```diff
@@ -0,0 +1,39 @@
+struct SpatialMassActionJump{T,S,U,V} <: AbstractMassActionJump
+    scaled_rates::T
```

Comment:
What about the hybrid case we discussed with both a vector and matrix for splitting up the reactions?

Reply:
Three cases we'd like `SpatialMassActionJump` to support:
1. `scaled_rates == Vector` and `scaled_spatial_rates == Nothing`.
2. `scaled_rates == Nothing` and `scaled_spatial_rates == Matrix`.
3. `scaled_rates == Vector` and `scaled_spatial_rates == Matrix`.
Note, we don't need to call the last field `scaled_spatial_rates`, that is just a placeholder. A more compact but representative name would be good.